python/ray/experimental/state.py

```diff
@@ -345,21 +346,101 @@ def task_profiles(self):
-    results = dict()
-    events = []
     for i in range(len(event_names)):
       event_list = self.redis_client.lrange(event_names[i], 0, -1)
       for event in event_list:
-        event_dict = json.loads(event.decode("ascii"))
+        event_dict = json.loads(event)
         task_id = ""
-        for element in event_dict:
-          if "task_id" in element[3]:
-            task_id = element[3]["task_id"]
-        if task_id != "":
-          if task_id not in results:
-            results[task_id] = []
-          results[task_id].append(event_dict)
-        else:
-          events.append(event_dict)
-    return results, events
+        for event in event_dict:
+          if "task_id" in event[3]:
+            task_id = event[3]["task_id"]
+        task_info[task_id] = dict()
+        for event in event_dict:
+          if event[1] == "ray:get_task" and event[2] == 1:
+            task_info[task_id]["get_task_start"] = event[0]
+          if event[1] == "ray:get_task" and event[2] == 2:
+            task_info[task_id]["get_task_end"] = event[0]
+          if event[1] == "ray:import_remote_function" and event[2] == 1:
+            task_info[task_id]["import_remote_start"] = event[0]
+          if event[1] == "ray:import_remote_function" and event[2] == 2:
+            task_info[task_id]["import_remote_end"] = event[0]
+          if event[1] == "ray:acquire_lock" and event[2] == 1:
+            task_info[task_id]["acquire_lock_start"] = event[0]
+          if event[1] == "ray:acquire_lock" and event[2] == 2:
+            task_info[task_id]["acquire_lock_end"] = event[0]
+          if event[1] == "ray:task:get_arguments" and event[2] == 1:
+            task_info[task_id]["get_arguments_start"] = event[0]
+          if event[1] == "ray:task:get_arguments" and event[2] == 2:
+            task_info[task_id]["get_arguments_end"] = event[0]
+          if event[1] == "ray:task:execute" and event[2] == 1:
+            task_info[task_id]["execute_start"] = event[0]
+          if event[1] == "ray:task:execute" and event[2] == 2:
+            task_info[task_id]["execute_end"] = event[0]
+          if event[1] == "ray:task:store_outputs" and event[2] == 1:
+            task_info[task_id]["store_outputs_start"] = event[0]
+          if event[1] == "ray:task:store_outputs" and event[2] == 2:
+            task_info[task_id]["store_outputs_end"] = event[0]
+          if "worker_id" in event[3]:
+            task_info[task_id]["worker_id"] = event[3]["worker_id"]
+          if "function_name" in event[3]:
+            task_info[task_id]["function_name"] = event[3]["function_name"]
+    return task_info
+
+  def actor_profiles(self):
+    actor_info = dict()
+    actors = self.redis_client.keys("Actor*")
+    for actor in actors:
+      actor_key_str = actor[len('Actor:'):]
+      actor_info['Actor:{}'.format(binary_to_hex(actor))] = self.redis_client.hgetall(actor)
+      x = actor_info['Actor:{}'.format(binary_to_hex(actor))]
+      if b'class_id' in x:
```

Comment:
We can safely assume that all of these fields are present, so we don't need the if statements. In fact, we could simplify this by doing something more like the following

```
... = {
    "class_id": binary_to_hex(x[b"class_id"]),
    "driver_id": binary_to_hex(x[b"driver_id"]),
    ...
}
```